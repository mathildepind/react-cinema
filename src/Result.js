import React from 'react';
import MovieDetails from './MovieDetails';

class Result extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      movieTitle : ''
    }
  this.handleMovieClick = this.handleMovieClick.bind(this);
  }


  handleMovieClick(event){
    let title = this.props.movie.Title;
    this.setState = {movieTitle : title};
    this.props.receiver(title);
    console.log(title);
  }


  render() {
    const {movie} = this.props;
    return (
      <div onClick={this.handleMovieClick}>
        <h2>{movie.Title}</h2>
        <h3>{movie.Year}</h3>
        <img
          src={movie.Poster}
          alt={movie.Title}
        />

      </div>
    )
  }
}




export default Result;
