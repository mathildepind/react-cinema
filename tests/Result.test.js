import React from 'react';
import Result from '../src/Result';
import renderer from 'react-test-renderer';


describe('Result', () => {
  it('matches the snapshot', () => {
    const movie = {
      Title : 'Batman Begins',
      Year : '2005',
      Poster : "xyz"
    }
    const tree = renderer.create (<Result movie = {movie} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
