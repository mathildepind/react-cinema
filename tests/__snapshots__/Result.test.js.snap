// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result matches the snapshot 1`] = `
<div
  onClick={[Function]}
>
  <h2>
    Batman Begins
  </h2>
  <h3>
    2005
  </h3>
  <img
    alt="Batman Begins"
    src="xyz"
  />
</div>
`;
